### PR TITLE
Use RegEx to filter pipelines and clean up configuration dialog

### PR DESCRIPTION
**Pull request by flosell**
This PR implements the changes discussed in #55: 

* Use a configurable RegEx to filter which pipelines to view and pull from GoCD
* Introduce a new environment variable to set a default RegEx
* Change the configuration dialog to make the new behaviour explicit in the UI: 
p24LQs5v0r

Note: The changes in `GoService` might conflict with changes in #54 so maybe merge that one first and I'll update this PR afterwards.

### CODE REVIEW

**Comment by flosell** [comment]
Updated this PR with the just merged changes from #54 and cleaned up a little bit. 
From my perspective, this PR is ready to review and merge now.

File: client/components/ConfigurationDialog.jsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  regexChanged(e) {
+    this.setState({
+      filterRegex: e.target.value
+    })
+  }
+
+  togglePipeline(event) {
+    let pipelineName = event.target.value;
+    let pipelineEnabled = event.target.checked;
+    let newDisabledPipelines = new Set(this.state.disabledPipelines)
```
Comment (karmats) [inline client/components/ConfigurationDialog.jsx]: IE doesn't supports `new Set(entries)` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set#Browser_compatibility) and I don't have a polyfill. But on the other hand who uses IE nowadays? 🤷‍♂

File: server/services/GoService.js [hunk 2]
```diff
@@ -27,7 +29,7 @@ export default class GoService {
     this.testService = new GoTestService(this.goConfig);
 
     // Init db and settings
-    this.dbService = new DBService(conf.dbFilePath);
+    this.dbService = new DBService(new Datastore({ filename: conf.dbFilePath, autoload: true }));
```
Comment (karmats) [inline server/services/GoService.js]: Much better 👍

File: server/services/DBService.js [hunk 3]
```diff
@@ -1,21 +1,34 @@
...
     if (profile) {
-      return this._executeDbAction('findOne',{$and: [{settings: {$exists: true}}, {profile: profile}]});
+      resultPromise = this._executeDbAction('findOne',{$and: [{settings: {$exists: true}}, {profile: profile}]});
     } else {
-      return this._executeDbAction('findOne',{$and: [{settings: {$exists: true}}, {profile: {$exists: false}}]});
+      resultPromise = this._executeDbAction('findOne',{$and: [{settings: {$exists: true}}, {profile: {$exists: false}}]});
     }
+
+    return resultPromise.then((result) => {
+      // Convert settings that were stored when filterRegex was only used for the configuration UI but not for filtering
```
Comment (karmats) [inline server/services/DBService.js]: Nice! 🌟

File: client/components/ConfigurationDialog.jsx (hunk 1 above)
Comment (flosell) [reply client/components/ConfigurationDialog.jsx]: Ah, interesting :) Just pushed a commit to change it to plain old Array features. Not quite as elegant but still ok.

**Comment by karmats** [comment]
Thanks for this @flosell

**Comment by karmats** [comment]
Closed #55

**Pull request merged**